i18n(it): translate `core-concepts` (#4529)
withastro/docs: src/content/docs/it/core-concepts/astro-components.mdx

---

I **componenti Astro** sono gli elementi costitutivi di base di qualsiasi progetto Astro. Sono componenti di modello solo HTML senza runtime lato client. Puoi individuare un componente Astro dalla sua estensione di file: `.astro`.

I componenti Astro sono estremamente flessibili. Spesso, un componente Astro conterrà alcune **UI riutilizzabili sulla pagina**, come un'intestazione o una scheda del profilo. Altre volte, un componente Astro può contenere un frammento di codice HTML più piccolo, come una raccolta di tag `<meta>` comuni che semplificano il lavoro con la SEO. I componenti Astro possono contenere anche un intero layout di pagina.

La cosa più importante da sapere sui componenti Astro è che **non vengono renderizzati sul client**. Eseguono il rendering in HTML in fase di compilazione o su richiesta utilizzando il [rendering lato server (SSR)](/it/guides/server-side-rendering/). Puoi includere il codice JavaScript all'interno del frontmatter del tuo componente e tutto verrà rimosso dalla pagina finale inviata ai browser dei tuoi utenti. Il risultato è un sito più veloce, con zero impatto JavaScript aggiunto per impostazione predefinita.

> Quando il tuo componente Astro necessita di interattività lato client, puoi aggiungere [tag HTML standard `<script>`](/it/guides/client-side-scripts/) o [componenti UI Framework](/it/core-concepts /componenti-strutturali/#componenti-interattivi-idratanti).

Author: Should probably remain untranslated, fully, though...

```suggestion
Quando il tuo componente Astro necessita di interattività lato client, puoi aggiungere [tag HTML standard `<script>`](/it/guides/client-side-scripts/) o [componenti UI Framework](/it/core-concepts/framework-components/#hydrating-interactive-components).
```

Se scrivi HTML semplice qui, il tuo componente rendererà quell'HTML in qualsiasi pagina Astro in cui viene importata e utilizzata.

Tuttavia, [la sintassi del modello del componente Astro](/it/core-concepts/astro-syntax/) supporta anche le **espressioni JavaScript**, Astro [`<style>`](/it/guides/styling/#styling-in -astro) e tag [`<script>`](/it/guides/client-side-scripts/#using-script-in-astro), **componenti importati** e [**direttive speciali Astro** ](/it/reference/directives-reference/). I dati e i valori definiti nello script del componente possono essere utilizzati nel modello del componente per produrre HTML creato dinamicamente.

```astro title="src/components/MyFavoritePokemon.astro"
---
// Il tuo script del componente qui!
import Banner from '../components/Banner.astro';
import ReactPokemonComponent from '../components/ReactPokemonComponent.jsx';
const myFavoritePokemon = [/* ... */];
const { title } = Astro.props;
---
<!-- Commenti HTML supportati! -->
{/*Anche la sintassi dei commenti JS è valida! */}

<Banner />
<h1>Hello, world!</h1>

<!-- Utilizza proprietà e altre variabili dallo script del componente: -->
<p>{title}</p>

> <!-- Includi altri componenti del framework dell'interfaccia utente con una direttiva `client:` per idratare: -->

Reviewer: ```suggestion
<!-- Includi componenti di altri framework UI con una direttiva `client:` per idratare: -->
```

See my comment at line 347 in 
`src/content/docs/it/core-concepts/astro-components.mdx`

<ReactPokemonComponent client:visible />

> <!-- Mescola HTML con espressioni JavaScript, simili a JSX: -->

Author: Like it.

```suggestion
<!-- Usa espressioni JavaScript all'interno di tag HTML, come in JSX: -->
```

  <Logo/>
  <h1>{title}</h1>
  <slot /> <!-- i figli senza `slot` o con l'attributo `slot="default"` andranno qui -->
  <Footer />
  <slot name="after-footer"/> <!-- i figli con l'attributo `slot="after-footer"` andranno qui -->
</div>
```

Per inserire contenuto HTML in un particolare slot, utilizza l'attributo `slot` su qualsiasi elemento figlio per specificare il nome dello slot. Tutti gli altri elementi secondari del componente verranno inseriti nel valore predefinito (senza nome) `<slot />`.

```astro /slot=".*?"/
---
// src/pages/fred.astro
import Wrapper from '../components/Wrapper.astro';
---
<Wrapper title="Fred's Page">
  <img src="https://my.photo/fred.jpg" slot="after-header">
  <h2>All about Fred</h2>
  <p>Here is some stuff about Fred.</p>
  <p slot="after-footer">Copyright 2022</p>
</Wrapper>
```

Utilizza un attributo `slot="my-slot"` sull'elemento figlio che desideri passare a un segnaposto `<slot name="my-slot" />` corrispondente nel tuo componente.

Tieni presente che gli slot denominati devono essere figli immediati del componente. Non è possibile passare slot con nome attraverso elementi nidificati.

:::tip

> Gli slot con nome possono anche essere passati a [componenti del framework UI](/it/core-concepts/framework-components/)!

Reviewer: ```suggestion
Gli slot con nome possono anche essere passati a [componenti di altri framework UI](/it/core-concepts/framework-components/)!
```
See my comment below.

	<h1>Astro</h1>
</HomeLayout>
```

## Componenti HTML

Astro supporta l'importazione e l'utilizzo di file `.html` come componenti o l'inserimento di questi file nella sottodirectory `src/pages/` come pagine. Potresti voler utilizzare componenti HTML se stai riutilizzando il codice di un sito esistente creato senza un framework o se vuoi assicurarti che il tuo componente non abbia funzionalità dinamiche.

I componenti HTML devono contenere solo HTML valido e pertanto non dispongono delle funzionalità chiave dei componenti Astro:
- Non supportano frontmatter, importazioni lato server o espressioni dinamiche.
- Tutti i tag `<script>` vengono lasciati non raggruppati, trattati come se avessero `is:inline`.
- Possono solo [fare riferimento alle risorse che si trovano nella cartella `public/`](/it/core-concepts/project-structure/#public).

:::note
Un [elemento `<slot />`](/it/core-concepts/astro-components/#slot) all'interno di un componente HTML funzionerà come in un componente Astro. Per utilizzare invece l'elemento [HTML Web Component Slot](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot), aggiungi `is:inline` al tuo `<slot >` elemento.
:::

## Prossimi passi

> 📚 Scopri come utilizzare i [componenti del framework UI](/it/core-concepts/framework-components/) nel tuo progetto Astro.

Reviewer: I think here - and elsewhere - `framework components` means `other framework components`, like React, Svelte, etc.
Hence I would not use `componenti del framework` which makes you think to Astro itself.
I would use `componenti di framework UI`.

Author: Gonna do a check if there are other places...

```suggestion
📚 Scopri come utilizzare i [componenti di framework UI](/it/core-concepts/framework-components/) nel tuo progetto Astro.
```
